### PR TITLE
enable llava int4 inference with autoround format

### PR DESCRIPTION
**Pull request by WeiweiZhang1**

### CODE REVIEW

File: auto_round/auto_quantizer.py [hunk 1]
```diff
@@ -456,7 +456,7 @@ class StoreAttr(object):
 
     def _process_model_before_weight_loading(self, model: "PreTrainedModel", **kwargs):
         if model.__class__.main_input_name != "input_ids":
-            raise RuntimeError("We can only quantize pure text model.")
+            logger.warning("We can only quantize pure text models and certain types of multimodal models.")
```
Comment (wenhuach21) [inline auto_round/auto_quantizer.py]: unclear, text submodule? and better detail "certain types"

File: examples/multimodal-modeling/Llava/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ This document presents step-by-step instructions for auto-round.
 
 In this example, we introduce an straight-forward way to execute quantization on some popular multimodal models such as LLaVA. 
 
-Please note that LLAVA quantization is currently an **experimental feature** and does not yet support inference on various devices after export.
+Please note that LLAVA quantized model is currently only support inference with **auto_round** export format.
```
Comment (wenhuach21) [inline examples/multimodal-modeling/Llava/README.md]: remove the last export

Comment (wenhuach21) [reply examples/multimodal-modeling/Llava/README.md]: the accuracy is real or qdq. if not real, better report the real accuracy later

File: examples/multimodal-modeling/Llava/mm_evaluation/textvqa.py [hunk 3]
```diff
@@ -199,3 +193,70 @@ def calculate_accuracy(self, result_file = None):
...
+    import argparse
+    from llava.mm_utils import get_model_name_from_path
+    from llava.model.builder import load_pretrained_model
+    from transformers import AutoConfig
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model_name", default="liuhaotian/llava-v1.5-7b"
+    )
+    parser.add_argument(
+        "--base_model", default="liuhaotian/llava-v1.5-7b"
```
Comment (wenhuach21) [inline examples/multimodal-modeling/Llava/mm_evaluation/textvqa.py]: are model_name and base model both needed?

File: examples/multimodal-modeling/Llava/mm_evaluation/textvqa.py [hunk 4]
```diff
@@ -199,3 +193,70 @@ def calculate_accuracy(self, result_file = None):
...
+    )
+    parser.add_argument(
+        "--eval_bs", default=4,
+    )
+    parser.add_argument(
+        "--trust_remote_code", action='store_true',
+        help="Whether to enable trust_remote_code"
+    )
+    parser.add_argument(
+        "--eval-question-file", type=str,
```
Comment (wenhuach21) [inline examples/multimodal-modeling/Llava/mm_evaluation/textvqa.py]: why use - instead of _

File: examples/multimodal-modeling/Llava/mm_evaluation/textvqa.py [hunk 5]
```diff
@@ -199,3 +193,70 @@ def calculate_accuracy(self, result_file = None):
...
+    parser.add_argument(
+        "--eval-annotation-file", type=str
+    )
+    args = parser.parse_args()
+    s = time.time()
+    config = AutoConfig.from_pretrained(args.model_name, trust_remote_code=args.trust_remote_code)
+    if hasattr(config, "quantization_config"):
+        quantization_config = config.quantization_config
+        if "quant_method" in quantization_config and "auto-round" in quantization_config["quant_method"]:
+            from auto_round.auto_quantizer import AutoHfQuantizer
```
Comment (wenhuach21) [inline examples/multimodal-modeling/Llava/mm_evaluation/textvqa.py]: using "from auto_round import AutoRoundConfig" with version >0.3

File: examples/multimodal-modeling/Llava/README.md (hunk 2 above)
Comment (WeiweiZhang1) [reply examples/multimodal-modeling/Llava/README.md]: the real accuracy will be updated later

File: auto_round/export/export_to_autogptq/export.py [hunk 6]
```diff
@@ -248,3 +248,4 @@ def save(model: torch.nn.Module, save_dir: str, max_shard_size: str = "5GB", saf
 
 
 
+
```
Comment (wenhuach21) [inline auto_round/export/export_to_autogptq/export.py:254]: pbar.update(1)
@register_format("auto_gptq")
reformat the code

File: auto_round/export/export_to_autoround/export.py [hunk 7]
```diff
@@ -294,6 +294,7 @@ def wrapper(name):
             if config["bits"] > 8:
                 modules_to_not_convert.append(name)
         save_awq(model, output_dir, modules_to_not_convert=modules_to_not_convert)
+    return model
```
Comment (wenhuach21) [inline auto_round/export/export_to_autoround/export.py:299]: remove "##" before save

File: auto_round/export/export_to_autoround/export.py [hunk 8]
```diff
@@ -360,3 +361,4 @@ def save_awq(
     if hasattr(model, "config") and hasattr(model.config, "quantization_config"):
         with open(os.path.join(save_dir, config_file), "w", encoding="utf-8") as f:
             json.dump(quantization_config, f, indent=2)
+
```
Comment (wenhuach21) [inline auto_round/export/export_to_autoround/export.py:364]: why not returning the compressing model as you do for other formats

**Review by wenhuach21 — APPROVED** [review]
